mmv1/third_party/tgc_next/test/assert_test_files.go

```diff
@@ -344,27 +345,26 @@ func compareHCLFields(map1, map2, ignoredFields map[string]any) []string {
 			continue
 		}
 
-		if _, ok := map2[key]; !ok {
-			// If a field is an empty map in map1, it is possible that its children are in map2.
-			// In this case, the field is not missing in map2.
-			// map1 = {a: {}}, map2 = {a.b: {}}
-			if vMap, ok := val.(map[string]interface{}); ok {
-				childrenFound := false
-				if len(vMap) == 0 {
-					for key := range map2 {
-						if strings.HasPrefix(key, key) {
-							childrenFound = true
-							break
-						}
-					}
-				}
-				if !childrenFound {
-					missingKeys = append(missingKeys, key)
-				}
-			} else {
-				missingKeys = append(missingKeys, key)
+		if rVal := reflect.ValueOf(val); !rVal.IsValid() || rVal.IsZero() {
+			continue
+		}
+
+		if sVal, ok := val.(string); ok {
+			if bVal, err := strconv.ParseBool(sVal); err == nil && !bVal {
```

Comment:
I have a concern that it is not safe to ignore the field, if the field is string. 

In [this test](https://github.com/hashicorp/terraform-provider-google/blob/main/google/services/iamworkforcepool/resource_iam_workforce_pool_provider_test.go#L481), attribute_condition is a string with "false" value.


I think the safe way is to check the field's type in resource schema. If it is boolean, then it is safe to make the conversion. 
It seems a lot of work to do to check the type. Maybe it is OK to keep `tgc_test_ignore_extra` for the edge cases?

Reply:
We could convert every field to the correct type before putting it in the flattened map, which would remove the need for these conditionals.

Personally I think it's safe to ignore any value which evaluates to a zero value even if the original type was not a boolean, because it's unlikely that a resource would only have the value `"false"` for a string value and also unlikely that that field would fail to be converted to and from HCL.

Reply:
> We could convert every field to the correct type before putting it in the flattened map, which would remove the need for these conditionals.
> 
We need to know every field's type in resource schema if we do this?

Reply:
We can go with the current logic. Thanks.